gatereader/reader.py

```diff
@@ -0,0 +1,187 @@
+import serial
+import struct
+from enum import Enum
+from .utils.structparse import mystruct, t, STRUCT_FORMAT
+
+class PacketId(Enum):
+    INVALID_PACKET        = 0x00
+    ANSWER_TO_RESET       = 0x01
+    CONTINUE_BOOT         = 0x02
+    FIRMWARE_UPGRADE      = 0x03
+    FW_UPGRADE_READY      = 0x04
+    FW_WRITE              = 0x05
+    FW_WRITE_ACK          = 0x06
+    FW_UPDATE_FINISH      = 0x07
+    GET_STATUS            = 0x08
+    SET_LED               = 0x09
+    ACK                   = 0x0A
+    BEEP                  = 0x0B
+    RFID_SEND             = 0x0C
+    RFID_SEND_COMPLETE    = 0x0D
+    RX_ERROR              = 0xFF
+
+class ResponseLength(Enum):
+    RESPONSE_ATR          = 5
+    TRANSMIT_OVERHEAD     = 3
+
+class Reader:
+
+    class ResetException(Exception):
```

Comment:
For all of these embedded classes: Don't put them inside `Reader` if they are ever meant to be used outside of `Reader`. (You'd have to do stuff like `reader.Reader.WhateverException` with qualified imports, which is horrible.)